main.go

```diff
@@ -146,10 +149,50 @@ func main() {
 			}
 		}
 
-		// and swap
 		scrapers = newScrapers
 
 		<-tick
 	}
+}
+
+func scrapeDockerVersion(cli *client.Client) {
+	tick := time.Tick(*interval)
+	log.Printf("Starting docker version scraper")
+
+	for {
+		server, err := cli.ServerVersion(context.Background())
+		if err != nil {
+			panic(err)
+		}
+		//log.Printf("version: %s", server.Version)
+		scrubbedVersion := versionScrubber.ReplaceAllString(server.Version,"")
+		dockerVersionFloat, err := strconv.ParseFloat(scrubbedVersion, 64)
+		if err != nil {
+			panic(err)
+		}
+		dockerVersion.With(prometheus.Labels{"docker_version": server.Version}).Set(dockerVersionFloat)
```

Comment:
as docker version can be string like: `17.09.3-ce` which is not float, I suggest to use docker version as a label value, and values set to 1. In this case you will be able to filter out by version as well as have any version without modifications. This will also make code a bit cleaner.

Reply:
The non-digits are scrubbed out immediately above. Code is tested. The point of this was not to provide anything human readable, but to make it easy to sort the docker versions numerically. I don't think there's a cost or a risk in having it be there. I've added appropriate logging, just in case.